fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkcommon/BulkDataUtils.java

```diff
@@ -137,21 +137,21 @@ public static void listBuckets(AmazonS3 cosClient) {
     private static int getFhirResourceFromBufferReader(BufferedReader resReader, int numOfLinesToSkip, List<Resource> fhirResources) throws Exception {
         int exported = 0;
         int lineRed = 0;
-        while (resReader.ready()) {
-            String resLine = resReader.readLine();
-            lineRed++;
-            if (resLine == null) {
-                break;
-            }
-            if (lineRed <= numOfLinesToSkip) {
-                continue;
-            }
-            fhirResources.add(FHIRParser.parser(Format.JSON).parse(new StringReader(resLine)));
-            exported++;
-            if (exported == Constants.IMPORT_NUMOFFHIRRESOURCES_PERREAD) {
-                break;
+        String resLine = null;
+        do {
+            resLine = resReader.readLine();
+            if (resLine != null) {
+                lineRed++;
+                if (lineRed <= numOfLinesToSkip) {
```

Comment:
remind me why we're skipping lines again?

Reply:
we support either reuse the inputstream or not reuse the inputstream, it's controlled by a switch, if we don't reuse the input stream, then the skipnumber will not be 0.    currently, I set the switch to on to reuse the inputstrean across all the chunk processing.  I could remove the "not reuse" codes if I find in the future that they are not really needed any more.